src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,935 @@
+                            audioChannelStats.setJitterTx(getJsonProperty(jsonNode, "jitter").floatValue());
+                            audioChannelStats.setPacketLossTx(getJsonProperty(jsonNode, "packetLoss").asInt());
+                            audioChannelStats.setPercentPacketLossTx(getJsonProperty(jsonNode, "percentPacketLoss").floatValue());
+                            audioChannelStats.setCodec(getJsonProperty(jsonNode, "mediaAlgorithm").asText());
+                            break;
+                        case "VIDEO":
+                            videoChannelStats.setBitRateTx(getJsonProperty(jsonNode, "actualBitRate").asInt());
+                            videoChannelStats.setJitterTx((getJsonProperty(jsonNode, "jitter").floatValue()));
+                            videoChannelStats.setPacketLossTx(getJsonProperty(jsonNode, "packetLoss").asInt());
+                            videoChannelStats.setPercentPacketLossTx(getJsonProperty(jsonNode, "percentPacketLoss").floatValue());
+                            videoChannelStats.setFrameRateTx(getJsonProperty(jsonNode, "actualFrameRate").floatValue());
+                            videoChannelStats.setCodec(getJsonProperty(jsonNode, "mediaAlgorithm").asText());
+                            videoChannelStats.setFrameSizeTx(getJsonProperty(jsonNode, "mediaFormat").asText());
+                            break;
+                        default:
+                            if(logger.isDebugEnabled()) {
+                                logger.debug("Not implemented for media type: " + getJsonProperty(jsonNode, "mediaType").asText());
+                            }
+                            break;
+                    }
+                    break;
+                default:
+                    if(logger.isDebugEnabled()) {
+                        logger.debug("Not implemented for media direction: " + getJsonProperty(jsonNode, "mediaDirection").asText());
+                    }
+                    break;
+            }
+        });
+
+        callStats.setTotalPacketLossTx(sumPacketData(audioChannelStats.getPacketLossTx(), videoChannelStats.getPacketLossTx()));
+        callStats.setTotalPacketLossRx(sumPacketData(audioChannelStats.getPacketLossRx(), videoChannelStats.getPacketLossRx()));
+        callStats.setPercentPacketLossTx(sumPacketData(audioChannelStats.getPercentPacketLossTx(), videoChannelStats.getPercentPacketLossTx()));
+        callStats.setPercentPacketLossRx(sumPacketData(audioChannelStats.getPercentPacketLossRx(), videoChannelStats.getPercentPacketLossRx()));
+
+        Integer callRateRx = sumPacketData(videoChannelStats.getBitRateRx(), audioChannelStats.getBitRateRx());
+        callStats.setCallRateRx((callRateRx == null) ? null : (callRateRx / 2));
+
+        Integer callRateTx = (sumPacketData(videoChannelStats.getBitRateTx(), audioChannelStats.getBitRateTx()));
+        callStats.setCallRateTx((callRateTx == null) ? null : (callRateTx / 2));

```

Comment:
I had a talk with Coco about it and the response was "default to video, but the callstatistics could be the sum of both" - does this make any sense?

Reply:
The logic here is not clear and does not looks correct. While theoretically call rate is the sum of media channel rates, the exact value at any given moment of time might differ. It is often being reported by devices as separate call rate property, if not - might be calculated, but has to be calculated correctly. In general, the rule of thumb for any monitored property is if it is reported by API we use it, if not - we report null and do not invent it, except from the cases when it can be calculated from other properties with reasonable reliability. And of course it has to be documented which is not here! Guys, sorry, but I'm getting fed up with reverse engineering your code, I'll stop approving any pull request the moment I reach a not obvious and not documented piece of code.
In this case, why the call rate is divided by 2?